tardis/montecarlo/src/cmontecarlo.c

```diff
@@ -489,7 +504,37 @@ void
 move_packet_across_shell_boundary (rpacket_t * packet,
                                    storage_model_t * storage, double distance, rk_state *mt_state)
 {
-  move_packet (packet, storage, distance);
+  // move the packet
+  uint64_t cur_shell = rpacket_get_current_shell_id (packet);
+  uint64_t next_shell = cur_shell + rpacket_get_next_shell_id (packet);
+  double new_r = rpacket_get_next_shell_id (packet) > 0 ?
+      storage->r_outer[cur_shell] :
+      storage->r_inner[cur_shell];
+  double doppler_factor = rpacket_doppler_factor (packet, storage);
+  distance = rpacket_get_d_boundary (packet);
+  rpacket_set_mu (packet,
+          (rpacket_get_mu (packet) * rpacket_get_r (packet) + distance) / new_r);
+  rpacket_set_r (packet, new_r);
+
+  // increment estimators
+  if (rpacket_get_virtual_packet (packet) <= 0)
+    {
+      double comov_energy = rpacket_get_energy (packet) * doppler_factor;
+      double comov_nu = rpacket_get_nu (packet) * doppler_factor;
+#ifdef WITHOPENMP
+#pragma omp atomic
+#endif
+      storage->js[rpacket_get_current_shell_id (packet)] +=
+          comov_energy * distance;
+#ifdef WITHOPENMP
+#pragma omp atomic
+#endif
+      storage->nubars[rpacket_get_current_shell_id (packet)] +=
+          comov_energy * distance * comov_nu;
+    }
+  // done moving packet
+
+  // reset tau event
```

Comment:
I asked myself the same question and came to the conclusion that `move_packet` and `move_packet_across_boundary` are two different functions which happen to have some bits in common.

- `move_packet` just moves the packet a known distance but 
- `move_packet_across_boundary` moves the packet TO a known point.
To me that's two different things. In theory the right thing to do would be to calculate `distance2boundary` again but we know that `d_boundary` didn't change between the last calculation and calling `move_packet_across_boundary` so I left it out.
Moving the packet and then setting the radius explicitly would work of course but that would repeat the calculation while using existing code. So both variants have their trade-offs.

The best solution in my opinion would be to have a function `increment_estimators` which is called by both methods. This would keep the (different) logic separated but combine the common one.